Kernel/Process.cpp

```diff
@@ -1183,6 +1209,17 @@ int Process::exec(String path, Vector<String> arguments, Vector<String> environm
     else if (elf_result.error().is_error())
         return elf_result.error();
 
+    if (set_in_exec()) {
```

Comment:
Why is this necessary? Why would we re-enter `do_exec`?

Reply:
We won't re-enter exec, but maybe a thread calls exec() at the same time as another (and I used to get a crash when that happened). But I removed it anyway, in the last version, fixed it another way (I'll comment in the PR directly)